libraries/Microsoft.Bot.Schema/ActivityEx.cs

```diff
@@ -507,6 +507,19 @@ public Activity ApplyConversationReference(ConversationReference reference, bool
             return this;
         }
 
+        /// <summary>
+        /// Determine if the Activity was sent via an Http/Https connection or Streaming
+        /// This can be determined by looking at the ServiceUrl property:
+        /// (1) All channels that send messages via http/https are not streaming
```

Comment:
Do we allow http channels? Other than localhost, I don't think we do.

Reply:
Most channels reply with https for a service url. Local will be http. Streaming won't be either of those.